core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java

```diff
@@ -346,7 +346,7 @@ public void testSimple_recursionTest() throws Exception {
         tag.setName("myname");
         tag.setValue("%{foo}");
         tag.setSize("10");
-        tag.setDynamicAttribute(null, "anotherAttr", "%{foo}");
+        tag.setDynamicAttribute(null, "anotherAttr", "another_%{foo}");
```

Comment:
That's intentional by me to start value with %{ to always have a monitor and test for double evaluation. This change eliminate that I think. Why had you to change it? to pass that test?

Reply:
Thanks for clarification! Just wanted to note and clarify it. So we can go ahead :) 

p.s. however I think it's a good practice to not touch current tests and just add tests, just for more confident that changes aren't breaking previous assertions. e.g. in this case I think we can simply add a new attr **anotherAttr2**:
```tag.setDynamicAttribute(null, "anotherAttr2", "another_%{foo}");```
and add expected attr to the current expected file. I see your point that dynAttr are normally `static_text_%{variable}` but this test is just a special not normal case test which assumes that attacker had been successful to inject such value.

Reply:
Sure, I can do it, no problem :)

Reply:
Done!

Reply:
thanks so much! LGTM :+1: will wait a few days and merge